sdk/tools/azure-openrewrite/src/main/resources/META-INF/rewrite/migrationRecipes/core/azure-core-migration.yml

```diff
@@ -0,0 +1,142 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------
+# Migration Recipes for Azure-Core Library
+# --------------------------------------------------------------------
+type: specs.openrewrite.org/v1beta/recipe
+name: com.azure.openrewrite.recipes.migrateAzureCore
+displayName: Migrate from azure-core to next generation stack
+description: This recipe migrates the Azure Storage Core library to the next generation stack.
+recipeList:
+  # Before: import com.azure.core.http.HttpHeaderName;
+  # After: import io.clientcore.core.http.models.HttpHeaderName;
+  - org.openrewrite.java.ChangeType:
+      oldFullyQualifiedTypeName: com.azure.core.http.HttpHeaderName
+      newFullyQualifiedTypeName: io.clientcore.core.http.models.HttpHeaderName
+
+  # Before: import com.azure.core.util.logging.ClientLogger;
+  # After: import io.clientcore.core.util.ClientLogger;
+  - org.openrewrite.java.ChangeType:
+      oldFullyQualifiedTypeName: com.azure.core.util.logging.ClientLogger
+      newFullyQualifiedTypeName: io.clientcore.core.instrumentation.logging.ClientLogger
```

Comment:
Doesn't this need further handling? There isn't a one-to-one API match between the ClientCore ClientLogger and azure-core ClientLogger

Reply:
@alzimmermsft Correct, the APIs need to be ported 100% not just the namespaces. That will require additional recipes to map anything that was previously publicly exposed to new APIs. This PR is to establish the testing framework and is not an indication that there is complete core support for migration.